### DISCUSSION TITLE
State of the Site: 2021

### DISCUSSION THREAD

**Discussion by mdryden** [110yards/fantasy#80] [Announcements]
Thanks to everyone who has helped out with donations and with issue reports, I think that we’ve had a great season on the whole.  The site has been overhauled twice since the original public launch, and I’m pretty happy with the results, despite the fact that there were a few issues I couldn’t easily fix in the middle of the season.

Now that the season is nearly over, I thought I would share some updates on how the season went, and start the planning for next year.

It is my intention to keep more things out in the open going forward, and to allow the community to help shape the direction the site takes in the future.

### The good

Huge support from the community of players.  A donation link was added to the site around week 3, and the flood of donations was incredible.  I received a total of $430.00 in donations, which reduced to $390.40 after Stripe processing fees.  As of November 8th, hosting has cost a total of $212.18.  Based on the current monthly average and adjusting for the upcoming end of the season, I expect this to end up around $260 for the full year.  **I will be donating the leftover money to The Canadian Amateur Football Association**.  I hope this will encourage players to continue to donate and support the site next year.  

**Thank you to everyone who donated this year**.

We had 25 active leagues this year, which I think is pretty good for a late launch and a mixed up season.  I hope to see this rise in 2022, but it wasn’t a bad thing to have a manageable number after a major site rewrite.

There was no major downtime.  There were a few incidents which required a bit of frantic work on my part, but for the most part, the issues were minor and easily resolved between weeks.  I’m happy to say that there were no notable outages this season.

### The bad

Hosting costs were quite a bit higher than I'd hoped. Goal for next season: reduce hosting costs by improving efficiency.  Donations covered it all for 2021 which is fantastic, but I'm not sure I can count on that forever.  Either way, it should be possible to bring the costs down; I have several changes in mind which I think will help.

Bugs which lasted all season.  There were a number of minor bugs which I wasn’t able to find a good solution to while ensuring the site remained stable.  For anyone who had a match affected by one of these bugs, I’m very sorry; I will do my best to fix these and keep that from happening in the future.

### What’s next in 2022?

Open source: I will be open sourcing the site for 2022.  This project is not a money maker, it’s a labour love and a programming playground for me.  The terms of my agreement with the CFL (which supplies the data that runs this site) do not allow me to make money from the use of their API, so it will never a commercially viable project.  Therefore, I think the best path forward is for me to give the source code to the community.  Perhaps others will get value seeing what I’ve done with the CFL’s API, or be interested in contributing new features to the site.  If you are interested in contributing, either with code or by testing, please don’t hesitate to reach out.

GitHub Discussions: I have enabled the new "Discussions" feature on this page, which I think will be useful in sharing knowledge around the community.

### Bug fixes

These are the top bugs I have on my list for 2022; I will be spending some time immediately after the season ends to resolve them.
Players not properly associated with a team
Season stats disappearing when a player goes on IL
Site briefly stops responding during game updates

### Site improvements

Efficiency improvements to reduce hosting costs.
Improve documentation for players, especially around system events like waivers and player locks.
Add “league history” for leagues which continue from year to year

### And finally… new features

I hope to add more functionality to the site this offseason.  Here are some of the requests which caught my eye, or that I thought of myself:

Trading - https://github.com/mdryden/110yards/issues/18
Re-order the “My Team” page - https://github.com/mdryden/110yards/issues/27
Draft history - https://github.com/mdryden/110yards/issues/37
Prevent roster moves with an ineligible roster

Please feel free to suggest additional features if you have any more ideas.  I can’t make any promises, but I’ll try to accommodate as many requests as I can.  That said, bugs and stability will always take priority over new features.

Thanks again to everyone who participated this year; I hope to see everyone back next season.

**Comment by mdryden** [comment]
Other suggestions:
- duplicate players (league option to allow more player choice)
- punter position
- support for stats which don't appear in the game boxscores

**Comment by bohsask** [comment]
On behalf of our league, thanks again for creating 110yards.   The site worked great, in our opinion. A very effective site overall for what really was a prototype product, and I think your offseason gameplan makes a tonne of sense.  

I would maybe add #71 (problem with scores displaying) to the high priority bug fix list.  Also, I know the reason why you limited RBs and QBs to one per roster was for gameplay reasons, but I think making these positions configurable as with the other roster spots would be a good enhancement, and let the leagues themselves decide on roster makeup.  We likely would have allowed backups on the same team to be rostered/protected if it had been possible.   

Bench limits by position would be another nice feature we talked about implementing as a league.  Which I believe was available back in the fantaseh days.   What we found this year with the open benches and no QB/RB allowed is that they basically ended up stacked with WR's and LB's, making the FA pool at those positions very thin  when combined with the bye week slots.    

Finally, I think with a bit of free marketing in the offseason: CFL fan board postings, twitter, etc, you could really drum up interest in this product.   With having a relationship/agreement with the CFL, I wonder as well if the CFL itself or TSN would help promote the option of season-long fantasy?

**Comment by SteveTaskersSon** [comment]
On behalf of all the guys in our league, a huge thank you for hosting CFL fantasy football on this site.  You have done a great job with it!  Our group of guys have been doing CFL fantasy for close to 15 years, most of the time using an excel spreadsheet to manage it.  Needless to say, your site has made life much easier and enjoyable...we'll definitely be back next year.    

I agree with bohsask's suggestions.  We also had the same issues re QB/RB/K roster limits and stacking WR on benches.  
A couple other suggestions we have: 
-allowing points for defensive touchdowns
-awarding different point values for different field goals lengths.  For example, field goals of 50+ could be worth 5 points, 35-49 yards 4 points, 0-34 yards 3 points...or something along those lines. 

Thanks again!

**Comment by kamikazecanuck** [comment]
Thanks again!  This was the best edition of 110 yards yet.  I think you've got some good ideas there.  
One thing I've been thinking of is that O-Flex should also allow QBs and well as RBs.  (Maintain hard cap on these positions or make it customizable).  One thing that happened this year is that some of the Qbs were so frikking crappy that I think some people would have rather just gone with another WR instead.

**Comment by MBControl-yyc** [comment]
First of all, thank you very much for all the effort.  

You and others have addressed many of the features that would be good fixes or additions, and I'm excited to see what is in store for 2022.  I'm also extremely happy to see the Canadian Amateur Football Assoc.  benefit from any surplus donations, and speaking for myself and the rest of my league, we will happy to continue to support you and the CAFA going forward.

My league is fairly small and casual, and we can't guarantee 8 teams, or an even number of teams each season.  In cases like these, we have created a "bye team" to fill out the schedule.  It would nice to have a possible schedule that would include bye weeks, in the cases where a league can't find an even number.  An auto draft feature would help as well.

We run a Canadian / National ratio (2 every week) in our line-ups, which is generally easy to track manually, but if there was a built in option that would track that (Illegal line-up warning), that'd be cool.

Lastly, I would really like to see an open text field on the league pages, where league messages, reminders and general smack talk can take place. 

My league and I will be discussing thoughts to participate in the evolution of this site, but we have been very happy with the progress to date, and hope to contribute were we can.

**Reply by mdryden** [reply]
Thanks for the kind words @mbcontrol-yyc.

Ratio options is something others have asked for as well, I'll put it on the issue list.  I think it should be fairly easy to accommodate.

An uneven team count is probably something I can accommodate too, but I'll probably put that lower on the list.

One option you could explore if I don't get to it is to simply add an extra team and leave the roster empty.  It would lose every week, but as long as you generated a schedule with the right number of weeks to make sure everyone played the bye team once, it would be pretty much the same as a bye.

**Comment by MBControl-yyc** [comment]
That's what we ended up doing, except we drafted a bad team for the bye
week team so people would still have to make some effort.

Appreciate you getting back to me, and we're looking forward to next year!

**Comment by mdryden** [comment]
@MBControl-yyc

I created issue #121 for the ratio enhancement - mind popping over there and weighing in on the behavior?  I have some questions / concerns on how to implement this one.

@kamikazecanuck  I think maybe you asked about this once too, so I'll tag you as well.

**Comment by MBControl-yyc** [comment]
Thank you sir.

We require 2 nationals per week, but it might be worth allowing leagues to
choose the number in their settings,  secondly, prompting a line-up error
message if minimum ratio is not met.  I would not have this stop a player
from submitting the line-up, as rosters can be changed throught the week,
including between games.

That's my 2 cents.
